### PR TITLE
Copyediting and some content adjustment & insertion

### PR DESCRIPTION
**Pull request by rvagg**
@yosuke-furukawa I've done some pretty heavy editing here, I know it deviates from your style somewhat and you're welcome to push back on anything you don't like. I've also added some additional content and edited some things that were inaccurate (as far as I know).

/cc @bmeck @chrisdickinson @fishrock123 - a scan for inaccuracies would be appreciated, this will hopefully go up on the NodeSource blog, I think we're all starting to feel the pressure to communicate better on this topic.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -332,24 +326,21 @@ bar.default(); // 2 ( assigned default function )
...
 
-Sample 1, using export default:
+**Example 1: Using `export default`**
 
 ```javascript
 // es.mjs
 let foo = {bar:'my-default'};
-// note:
 export default foo;
 foo = null; // this null value does not effect import value.
```
Comment (bmeck) [inline README.md:345]: we should probably also include a live bound form here. as people may think we don't do live binding.

Comment (rvagg) [inline README.md:345]: As in, exporting an object that can change over its lifetime? `foo = { counter: 1 }` perhaps

Comment (bmeck) [inline README.md:345]: `export default foo` ~= `export let default = foo`

It is hard to represent the "live binding" of things using `default`.

Having a 2nd export might help explain that live bindings still work.

``` js
// export the *variable* foo
export let foo = {bar:'my-default'};

// note: this exports the current *value* of foo
export default foo;

// this null value does not affect the default export
foo = null; 
```

Comment (rvagg) [inline README.md:345]: ah, good example

**Comment by yosuke-furukawa** [comment]
I have read all your changes, mostly LGTM. 
I think the article is a bit old, node-eps draft is changed. 
Could I fix some new diagrams and algorithms?

https://github.com/rvagg/esmodules_on_node/blob/255b8510f4fac091cad50cd4ad0573a0c572af54/README.md#basic-interoperability-algorithm

**Comment by rvagg** [comment]
@yosuke-furukawa absolutely, merge this PR and go ahead and edit. I did try and bring some of it up to date already but there's so much going on in this area that my head isn't wrapped around that I'm sure it's not 100%.

**Comment by yosuke-furukawa** [comment]
@rvagg merged. I am fixing the algorithm part. but I think current algorithm is not so different from old one.

**Pull request merged**